openquake/calculators/event_based.py

```diff
@@ -101,13 +103,45 @@ def event_based(proxies, full_lt, oqparam, dstore, monitor):
                     continue
                 proxy.geom = rupgeoms[proxy['geom_id']]
                 ebr = proxy.to_ebr(cmaker.trt)  # after the geometry is set
-                try:
-                    computer = GmfComputer(
-                        ebr, srcfilter.sitecol.filtered(sids), cmaker,
-                        oqparam.correl_model, oqparam.cross_correl,
-                        oqparam._amplifier, oqparam._sec_perils)
-                except FarAwayRupture:
-                    continue
+                if "station_data" in oqparam.inputs:
+                    station_sites = dstore.read_df('station_sites')
+                    station_data = dstore.read_df('station_data')
+                    station_sites = SiteCollection.from_points(
+                        lons=station_sites.lon.values,
+                        lats=station_sites.lat.values,
+                    )
+                    station_sitemodel = station_sites.assoc(sitecol, assoc_dist=None)
+                    station_sitecol = SiteCollection.from_points(
+                        lons=station_sites.lon,
+                        lats=station_sites.lat,
+                        sitemodel=station_sitemodel,
+                    )
+                    stnfilter = SourceFilter(
+                        station_sitecol, oqparam.maximum_distance(trt))
+                    stnids = stnfilter.close_sids(proxy, trt)
+                    if len(stnids) < len(station_sites):
+                        logging.warning('%d stations filtered away', len(station_sites) - len(stnids))
+                    if len(stnids) == 0:  # all stations filtered away
+                        continue
+                    try:
+                        computer = ConditionedGmfComputer(
+                            ebr, srcfilter.sitecol.filtered(sids), 
+                            stnfilter.sitecol.filtered(stnids), 
+                            station_data.loc[stnids], oqparam.observed_imts,
+                            cmaker, oqparam.correl_model, oqparam.cross_correl,
+                            oqparam.ground_motion_correlation_params,
+                            oqparam.number_of_ground_motion_fields,
+                            oqparam._amplifier, oqparam._sec_perils)
+                    except FarAwayRupture:
```

Comment:
I guess we should log a warning here, as done before about filtered away stations